### DISCUSSION TITLE
How to keep the kafka / Zk / MM2 undeployed

### DISCUSSION THREAD

**Discussion by palpandy** [strimzi/strimzi-kafka-operator#10402] [General]
Hello, 

Is there any way to keep the kafka / zk / mm2 cluster undeployed ? I tried setting the replicas to 0 but it didn't work because strimzi doesn't allow it. The reason why I am asking is I want to keep the cluster as Passive cluster so that the Active cluster goes down , producers / consumers can connect to the passive cluster which will be deployed immediately. My goal is to keep the kafka CR ready but it shouldn't get deployed by Strimzi and whenever I want to switch , the cluster has to get deployed with a little changes on the CR

Please advice if there is any way to do it

**Comment by scholzj** [comment]
You should probably clarify what exactly you mean by undeployed. But I do not think it works the way you expect it to work. The MM2 as well as the passive cluster in the active-passive cluster pair still needs to be running and syncing the data. The passive cluster is passive only in the sense that it is not used by clients. If the MM2 and the passive cluster would not be running, they won't have any messages or topics when the active cluster fails and you would need to switch to them.

**Reply by palpandy** [reply]
Hi scholzj,

Basically, I want to keep the configs [CR] ready for the kafka cluster but it should not be deployed . So, I tried setting replicas to 0 for kafka / zk CR just to keep it ready so that whenever I need this cluster it can de deployed by changing only the replicas value to 1 or more. But, this doesn't work because I got the error saying minimum replicas should be 1 or more if i try to apply the CR.

**Reply by scholzj** [reply]
I guess you can create them with the paused reconciliation annotation. Then the operator will ignore the custom resource. You can also simply store them in Git and use some GitOps mechanism to create them only when needed.

**Reply by palpandy** [reply]
okay this helps. Thanks for the support scholzj

**Reply by palpandy** [reply]
I just need one more help. Is there any  way we can undeploy [delete] the cluster by changing some values [may be replica values or something] in kafka CR or in strimzi apart from removing CR manually and deleting the cluster ?

**Reply by scholzj** [reply]
No, you cannot do that. You can delete the custom resource. As I said, from what you are describing, the right way is to use some GitOps mechanism to create / delete the custom resources.

**Reply by palpandy** [reply]
ok thanks
